### PR TITLE
Add test for directly adding hidden file

### PR DESCRIPTION
**Pull request by Kubuxu**
License: MIT
Signed-off-by: Jakub Sztandera <kubuxu@protonmail.ch>

### CODE REVIEW

File: test/sharness/t0040-add-and-cat.sh [hunk 1]
```diff
@@ -30,6 +30,14 @@ test_add_cat_file() {
         echo "added $HASH hello.txt" >expected &&
     	test_cmp expected actual
     '
+    test_expect_success "ipfs add on hidden file succeeds" '
+    	echo "Hello Worlds!" >mountdir/.hello.txt &&
+        ipfs add mountdir/.hello.txt >actual
+    '
+
+    test_expect_success "ipfs add on hiddne file output looks good" '
```
Comment (dignifiedquire) [inline test/sharness/t0040-add-and-cat.sh]: typo. `hiddne`

**Comment by Kubuxu** [comment]
It was broken not that far into the past again (about 70 commits).

File: test/sharness/t0040-add-and-cat.sh [hunk 2]
```diff
@@ -30,7 +30,6 @@ test_add_cat_file() {
         echo "added $HASH hello.txt" >expected &&
     	test_cmp expected actual
     '
-
```
Comment (whyrusleeping) [inline test/sharness/t0040-add-and-cat.sh]: lets go ahead and leave this newline here, lol

Comment (Kubuxu) [inline test/sharness/t0040-add-and-cat.sh]: Missed it.

**Comment by Kubuxu** [comment]
Fixed it.

File: test/sharness/t0040-add-and-cat.sh [hunk 3]
```diff
@@ -77,6 +77,17 @@ test_add_cat_file() {
         echo "added $HASH hello.txt" >expected &&
         test_cmp expected actual
     '
+
+    test_expect_success "ipfs add on hidden file succeeds" '
+    	echo "Hello Worlds!" >mountdir/.hello.txt &&
+        ipfs add mountdir/.hello.txt >actual
```
Comment (chriscool) [inline test/sharness/t0040-add-and-cat.sh]: It looks like the line above is not properly indented.

File: test/sharness/t0040-add-and-cat.sh [hunk 4]
```diff
@@ -77,6 +77,17 @@ test_add_cat_file() {
...
+    '
+
+    test_expect_success "ipfs add on hidden file output looks good" '
+    	HASH="QmVr26fY1tKyspEJBniVhqxQeEjhF78XerGiqWAwraVLQH" &&
+        echo "added $HASH .hello.txt" >expected &&
```
Comment (chriscool) [inline test/sharness/t0040-add-and-cat.sh]: The line above also seems to have an indentation issue. Or perhaps it is the line above it?

**Comment by chriscool** [comment]
Otherwise this LGTM!

**Pull request merged**